bridge/client/app/_components/ktb-modify-uniform-subscription/ktb-modify-uniform-subscription.component.ts

```diff
@@ -0,0 +1,153 @@
+      switchMap(params => {
+        this.editMode = !!params.subscriptionId;
+        if (params.subscriptionId) {
+          return this.dataService.getUniformSubscription(params.integrationId, params.subscriptionId);
+        } else {
+          return of(new UniformSubscription(params.projectName));
+        }
+      }),
+      tap(subscription => {
+        this.taskControl.setValue(subscription.prefix);
+        this.taskSuffixControl.setValue(subscription.suffix);
+        this.isGlobalControl.setValue(subscription.isGlobal);
+      }),
+      take(1)
+    );
+
+    const integrationId$ = this.route.paramMap
+      .pipe(
+        map(paramMap => paramMap.get('integrationId')),
+        filter((integrationId: string | null): integrationId is string => !!integrationId),
+        take(1)
+      );
+
+    const projectName$ = this.route.paramMap
+      .pipe(
+        map(paramMap => paramMap.get('projectName')),
+        filter((projectName: string | null): projectName is string => !!projectName)
+      );
+
+    const taskNames$ = projectName$
+      .pipe(
+        switchMap(projectName => this.dataService.getTaskNames(projectName)),
+        take(1)
+      );
+    const project$ = projectName$
+      .pipe(
+        switchMap(projectName => this.dataService.getProject(projectName)),
+        filter((project?: Project): project is Project => !!project),
+        tap(project => this.updateDataSource(project)),
+        take(1)
+      );
+
+    this.data$ = forkJoin({
+      taskNames: taskNames$,
+      subscription: subscription$,
+      project: project$,
+      integrationId: integrationId$
+    });
+  }
+
+  private setValidation(): void {
+    this.taskControl.setValidators([Validators.required]);
+    this.taskSuffixControl.setValidators([Validators.required]);
+  }
+
+  private updateDataSource(project: Project): void {
+    this._dataSource.data = {
+      autocomplete: [
+        {
+          name: 'Stage',
+          autocomplete: project.stages.map(stage => {
+            return {
+              name: stage.stageName
+            };
+          })
+        },
+        {
+          name: 'Service',
+          autocomplete: project.getServices().map(service => {
+            return {
+              name: service.serviceName
+            };
+          })
+        }
+      ]
+    } as DtFilterFieldDefaultDataSourceAutocomplete;
+  }
+
+  public updateSubscription(projectName: string, integrationId: string, subscription: UniformSubscription): void {
+    this.updating = true;
+    let update;
+    subscription.event = `${EventTypes.PREFIX}${this.taskControl.value}.${this.taskSuffixControl.value}`;
+    subscription.setIsGlobal(this.isGlobalControl.value, projectName);
+
+    if (this.editMode) {
+      update = this.dataService.updateUniformSubscription(integrationId, subscription);
+    } else {
+      update = this.dataService.createUniformSubscription(integrationId, subscription);
+    }
+    update.subscribe(() => {
```

Comment:
This subscription will be potentially kept open. Unsubscribe onDestroy or `take(1)`

Reply:
This is an HTTP-Call, so it is like `take(1)`

Reply:
Ok :)